test/scripts/test.sh

```diff
@@ -1,122 +1,148 @@
 #!/bin/bash
 
-#set -e
-set -o pipefail
+set -e
+#set -o pipefail
 
 CMD=$1
-BUILDDIR=$2
+BUILDDIR="@CMAKE_BINARY_DIR@"
 WORKDIR=$BUILDDIR/tmptest
-EXTFILE=$BUILDDIR/*--*.sql
-SOFILE=$(echo "$BUILDDIR"/lib*.so)
-PSQL="psql -h $WORKDIR/lock -e --set ON_ERROR_STOP=0 postgres"
-FAILPSQL="psql -h $WORKDIR/lock -e --set ON_ERROR_STOP=1 postgres"
-DBDIR=$WORKDIR/db
-PGCTL="pg_ctl -w -D $DBDIR -l $WORKDIR/log/postgres.log -o -k -o $WORKDIR/lock -o -h -o ''" # -o -c -o enable_seqscan=off -o -c -o enable_bitmapscan=off -o -c -o enable_indexscan=on -o -c -o enable_indexonlyscan=on"
+# TODO this is currently fixed for linux flavors. Will not work on windows
+EXTFILE="$BUILDDIR/lib@MOBILITYDB_EXTENSION_FILE@.so"
+BIN_DIR="@POSTGRESQL_BIN_DIR@"
 
-#FIXME: this is cheating
-PGSODIR=$(pg_config --pkglibdir)
-POSTGIS=$(find "$PGSODIR" -name 'postgis-2.5.so' | head -1)
+SOFILE="$BUILDDIR/@MOBILITYDB_LIB_NAME@)"
+PSQL="${BIN_DIR}/psql -h $WORKDIR/lock -e --set ON_ERROR_STOP=0 postgres"
+FAILPSQL="${BIN_DIR}/psql -h $WORKDIR/lock -e --set ON_ERROR_STOP=1 postgres"
+DBDIR="$WORKDIR/db"
+
+PGCTL="${BIN_DIR}/pg_ctl -w -D $DBDIR -l $WORKDIR/log/postgres.log -o -k -o $WORKDIR/lock -o -h -o ''"
+# -o -c -o enable_seqscan=off -o -c -o enable_bitmapscan=off -o -c -o enable_indexscan=on -o -c -o enable_indexonlyscan=on"
+
+PGSODIR="@POSTGRESQL_DYNLIB_DIR@"
+POSTGIS="@POSTGIS_LIBRARY@"
 
 case $CMD in
 setup)
-	rm -rf "$WORKDIR"
-	mkdir -p "$WORKDIR"/db "$WORKDIR"/lock "$WORKDIR"/out "$WORKDIR"/log
-	initdb -D "$DBDIR" 2>&1 | tee "$WORKDIR"/log/initdb.log
-
-	if [ ! -z "$POSTGIS" ]; then
-		POSTGIS=$(basename "$POSTGIS" .so)
-		echo "shared_preload_libraries = '$POSTGIS'" >> "$WORKDIR"/db/postgresql.conf
-	fi
-	echo "max_locks_per_transaction = 128" >> "$WORKDIR"/db/postgresql.conf
-	echo "timezone = 'UTC'" >> "$WORKDIR"/db/postgresql.conf
-	echo "parallel_tuple_cost = 100" >> "$WORKDIR"/db/postgresql.conf
-	echo "parallel_setup_cost = 100" >> "$WORKDIR"/db/postgresql.conf
-	echo "force_parallel_mode = off" >> "$WORKDIR"/db/postgresql.conf
-	echo "min_parallel_table_scan_size = 0" >> "$WORKDIR"/db/postgresql.conf
-	echo "min_parallel_index_scan_size = 0" >> "$WORKDIR"/db/postgresql.conf
-
-	$PGCTL start 2>&1 | tee "$WORKDIR"/log/pg_start.log
-	if [ "$?" != "0" ]; then
-		sleep 2
-		$PGCTL status
-
-		if [ "$?" != "0" ]; then
-			echo "Failed to start PostgreSQL" >&2
-			$PGCTL stop
-			exit 1
-		fi
-	fi
-
-	exit 0
-	;;
+  rm -rf "$WORKDIR"
+  mkdir -p "${DBDIR}" "$WORKDIR"/lock "$WORKDIR"/out "$WORKDIR"/log
+  "${BIN_DIR}/initdb" -D "$DBDIR" 2>&1 | tee "$WORKDIR/log/initdb.log"
+
+  echo "POSTGIS = $POSTGIS" >> "$WORKDIR/log/initdb.log"
+  if [ -n "$POSTGIS" ]; then
+    POSTGIS=$(basename "$POSTGIS" .so)
+    echo "shared_preload_libraries = '$POSTGIS'" >> "${DBDIR}"/postgresql.conf
```

Comment:
Talking with @robe2, it is clear that this is a no-no.
Which reminds me that on the actions:
https://github.com/MobilityDB/MobilityDB/runs/3267819299?check_suite_focus=true
```
Run sudo service postgresql start
  sudo service postgresql start
  sudo -u postgres createdb -p ${PGPORT}  ___pgr___test___
  sudo -u postgres psql -p ${PGPORT}  -d ___pgr___test___ -c "CREATE EXTENSION mobilitydb CASCADE; SELECT mobilitydb_version()"
  shell: /bin/bash -e {0}
  env:
    PGVER: 13
    PGPORT: 5433
NOTICE:  installing required extension "postgis"
   mobilitydb_version    
-------------------------
 MobilityDB 1.0.0 Beta 3
(1 row)
```
The extension can be installed without having:
1) postgis preloaded
2) and the changes on the postgresql.conf

Of course, one thing is create the extension and the other is to use it
- A packager will never make those changes to postgres.conf.
```
$grep max_locks_per_transaction /etc/postgresql/12/main/postgresql.conf
#max_locks_per_transaction = 64         # min 10
```